src/gas/GasFeeController.ts

```diff
@@ -329,7 +329,9 @@ export class GasFeeController extends BaseController<
     this.EIP1559APIEndpoint = EIP1559APIEndpoint;
     this.legacyAPIEndpoint = legacyAPIEndpoint;
     this.clientId = clientId;
-    if (onNetworkStateChange && getChainId && getProvider) {
+    const initialProvider = getProvider();
```

Comment:
Nit: This could be moved into the `if` block as well. It is only used there. I guess it doesn't matter though.

Reply:
Done [here](https://github.com/MetaMask/controllers/pull/936/commits/2027577adde625d62babf7e183e95e866b3e596b)